tests/Support/SupportCollectionTest.php

```diff
@@ -1678,6 +1678,18 @@ public function testHas($collection)
         $this->assertFalse($data->has(['third', 'first']));
     }
 
+    /**
+     * @dataProvider collectionClassProvider
+     */
+    public function testDoesntHave($collection)
+    {
+        $data = new $collection(['id' => 1, 'first' => 'Hello', 'second' => 'World']);
+        $this->assertFalse($data->doesntHave('first'));
+        $this->assertTrue($data->doesntHave('third'));
+        $this->assertFalse($data->doesntHave(['first', 'second']));
+        $this->assertTrue($data->doesntHave(['third', 'first']));
```

Comment:
The `has` method returns `true` only if _all_ keys passed to it are present.

Shouldn't the `doesntHave` method also only return `true` if _all_ keys are missing?

Reply:
I'm not sure how to achieve the described behavior, can I ask for help?

Reply:
I've tried to do the following which should give the expected behavior but with no luck:
```php
<?php
    /**
     * Determine if an item does not exist in the collection by key.
     *
     * @param  mixed  $key
     * @return bool
     */
    public function doesntHave($key)
    {
        return ! $this->has($key);
    }
```

Reply:
I was thinking about this too, actually a better name for this should be something like hasMissing or similar.

This implementation checks till the first occurrence.

to check for all missing elements the condition of if needs to be negated, not the return value.
--
An other possible way would be to have a doesntHaveAny method implementing that feature.

I'm not sure about the correct terminology since English is not my native language:
- has: has all passed key
- doesntHave: has none of the keys
- hasAny: has one of the passed keys
- doesntHaveOne: missing at least one <-- Implemented by this PR

Reply:
Actually if this is used:
```php
<?php
    /**
     * Determine if an item does not exist in the collection by key.
     *
     * @param  mixed  $key
     * @return bool
     */
    public function doesntHave($key)
    {
        return ! $this->has($key);
    }
```
it works as expected a true inverse of the `has` method. Also I've updated the test with the following which should prove it:
```php
<?php
    /**
     * @dataProvider collectionClassProvider
     */
    public function testDoesntHaveReturnsValidResults($collection)
    {
        $data = new $collection(['foo' => 'one', 'bar' => 'two', 1 => 'three']);
        $this->assertFalse($data->doesntHave('foo'));
        $this->assertFalse($data->doesntHave('foo', 'bar', 1));
        $this->assertFalse($data->doesntHave('foo', 'bar', 1, 'baz'));
        $this->assertTrue($data->doesntHave('baz'));
        $this->assertTrue($data->doesntHave(['baz', '2', 3]));
    }
```

Reply:
@JosephSilber, @taylorotwell what do you think, should I make a new PR?

Reply:
ping @JosephSilber @taylorotwell